src/App.jsx

```diff
@@ -14,7 +14,7 @@ function App() {
   <Header/>
   <Routes>
     <Route path="/" element={<ArticleList articles={articles} setArticles={setArticles}/>}></Route>
-    <Route path="/articles/:article_id" element={<ArticlePage/>}></Route>
+    <Route path="/articles/:article_id" element={<ArticlePage articles={articles}/>}></Route>
```

Comment:
Ideally we should not have state held up in app as this causes the whole app to re-render and that is not ideal, especially when the app will include more paths and components that do not need this state.